students/shibata/session3/mailroom.py

```diff
@@ -0,0 +1,150 @@
+
+        #                        print()#adding a new line
+
+                    elif (strName.strip() == 'back'):
+                        break #and Exit the program
+                    else:
+                        for row in lstTable:
+
+        #                    print("list: " + row["Name"])
+                            if (strName == row["Name"]):
+                                print("name is in the list, use " + row["Name"] + " who already donated total $" + str(row["Total"]) + " of " + str(row["Num"]) + " donations")
+
+#                                floatTotal = float(input("What is the Total Given? - ")).strip()
+                                floatTotal = float(input("What is the Total Given? - "))
+
+                                floatTotal = float(floatTotal) + float(row["Total"])
+                                print(floatTotal)
+#                                intNum = int(input("How many Number of Gifts? - ")).strip()
+                                intNum = int(input("How many Number of Gifts? - "))
+                                intNum = int(intNum) + int(row["Num"])
+                                print(intNum)
+                                dicRow = {"Name":strName,"Total":floatTotal,"Num":intNum}
+                                lstTable.append(dicRow)
+#                                lstTable.append(dicRow)
+                                print("Donor Name | Total Given | Num Gifts | Average Gift\n------------------------------------------------------------------")
+                                for row in lstTable:
+
+                                                        #            Total = float(row["Total"])
+                                                        #            Num = int(row["Num"])
+                                                        #            Ave = (Total/Num)
+                                                        #            print(Ave)
+                                                        #            print(row["Name"] + '         $' + row["Total"] + row["Num"]  + '  $' + "{:.2f}".format(str(float(row["Total"])/int(row["Num"]))))
+                                                        #            print(row["Name"] + '         $' + row["Total"] + row["Num"]  + '  $' + "{:.2f}".format(float(row["Total"])/int(row["Num"])))
+                                                        #            print ("{}, ${:.2f}, {:.2f}, ${:.2f}".format(row["Name"], row["Total"], row["Num"], float(row["Total"])/int(row["Num"])))
+                                    print ("{}, ${:.2f}, {}, ${:.2f}".format(row["Name"], float(row["Total"]), row["Num"], float(row["Total"])/int(row["Num"])))
+                            break #to show the menu
+
+                        else:
+                            floatTotal = str(input("What is the Total Given? - ")).strip()
+                            intNum = str(input("How many Number of Gifts? - ")).strip()
```

Comment:
careful here -- again, I don't like names like "intNum", but certainly not for a string! WHynot call int() on it right here? And you don't need to callstr() at all.